velox/dwio/dwrf/common/Common.h

```diff
@@ -340,6 +340,189 @@ class PostScript {
   uint64_t stripeStatisticsLength_;
 };
 
+// TODO: wrap stripes, types, metadata, column statistics
+class Footer {
+ public:
+  explicit Footer(proto::Footer* footer)
+      : headerLength_{footer->has_headerlength() ? footer->headerlength() : 0},
+        contentLength_{
+            footer->has_contentlength() ? footer->contentlength() : 0},
+        numberOfRows_{footer->has_numberofrows() ? footer->numberofrows() : 0},
+        rowIndexStride_{
+            footer->has_rowindexstride() ? footer->rowindexstride() : 0},
+        rawDataSize_{footer->has_rawdatasize() ? footer->rawdatasize() : 0},
+        checksumAlgorithm_{
+            footer->has_checksumalgorithm() ? footer->checksumalgorithm()
+                                            : proto::ChecksumAlgorithm::NULL_},
+        dwrfFooter_{footer} {
+    stripeCacheOffsets_.reserve(footer->stripecacheoffsets_size());
+    for (const auto offset : footer->stripecacheoffsets()) {
+      stripeCacheOffsets_.push_back(offset);
+    }
+  }
+
+  const proto::Footer* getDwrfFooter() const {
```

Comment:
We will need to keep this in the future because our internal system relies on this proto object.  You can return `nullptr` if it is an ORC implementation.

Reply:
Sure.